libpf/pfelf/internal/mmap/mmap.go

```diff
@@ -22,11 +23,25 @@ var (
 // Like any io.ReaderAt, clients can execute parallel ReadAt calls, but it is
 // not safe to call Close and reading methods concurrently.
 type ReaderAt struct {
+	// refCount is the number of references
+	refCount atomic.Int32
+
 	data []byte
 }
 
+// Take takes a reference on the data
+func (r *ReaderAt) Take() io.Closer {
+	r.refCount.Add(1)
+	return r
+}
+
 // Close closes the reader.
 func (r *ReaderAt) Close() error {
+	// Drop reference
+	if r.refCount.Add(-1) != 0 {
```

Comment:
Nit and probably not needed here but safer:

```suggestion
	if r.refCount.Add(-1) > 0 {
```

Reply:
Resolving with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/456/commits/54f9da99896bd1aad4e19f651b37620bd5c42c64